### PR TITLE
update node sdk to v1.0.0

### LINKED ISSUE #10: The SDK doesn't support the "hosted_url" parameter returned in the response.
When creating a transaction, the documentation specifies we should be able to retrieve the 'hosted_url'. However, in the SDK, when calling the `createTransaction` method, it calls the `_fetchTokens` method and this method only returns a `Token` object. Therefore, we are unable to retrieve the `hosted_url`. The only way is to make the REST Api calls ourselves, which kind of defeats the purpose of having the SDK entirely.

See relevant code here (I suspect the fix would require quite a bit of a refactor):

```
Client.prototype._fetchTokens = async function (path, payload) {
  var headers = {
    Authorization:
      "Basic " + new Buffer.from(`${this.apiSecret}:`).toString("base64"),
    "User-Agent": "BerbixNode/" + SDK_VERSION,
  };
  var result = await this._request("POST", path, headers, payload);
  return new Tokens(
    result.refresh_token,
    result.access_token,
    result.client_token,
    now() + result.expires_in,
    result.transaction_id,
    result
  );
};

Client.prototype.createTransaction = function (opts) {
  var payload = {};
  if (opts.email != null) {
    payload.email = opts.email;
  }
  if (opts.phone != null) {
    payload.phone = opts.phone;
  }
  if (opts.customerUid != null) {
    payload.customer_uid = "" + opts.customerUid;
  }
  if (opts.templateKey != null) {
    payload.template_key = opts.templateKey;
  }
  if (opts.hostedOptions != null) {
    var hostedOptions = {};
    if (opts.hostedOptions.completionEmail != null) {
      hostedOptions.completion_email = opts.hostedOptions.completionEmail;
    }
    if (opts.hostedOptions.redirectUrl != null) {
      hostedOptions.redirect_url = opts.hostedOptions.redirectUrl;
    }
    payload.hosted_options = hostedOptions;
  }
  return this._fetchTokens("/v0/transactions", payload);
};
```

### PR DESCRIPTION
**Pull request by andrewsli**
Updates node sdk to 1.0

- removes deprecated methods
- removes ability to set environments, keep ability to specify api host
- add createHostedTransaction method

Closes #10

### CODE REVIEW

File: lib/berbix.js [hunk 1]
```diff
@@ -160,34 +167,15 @@ Client.prototype._fetchTokens = async function (path, payload) {
...
+  var payload = optionsToPayloadObject(opts)
   return this._fetchTokens("/v0/transactions", payload);
 };
 
-// createUser is deprecated, please use createTransaction
-Client.prototype.createUser = Client.prototype.createTransaction;
+Client.prototype.createHostedTransaction = async function(opts) {
+  var payload = optionsToPayloadObject(opts)
+  const tokens = await this._fetchTokens("/v0/transactions", payload)
+  return {tokens, hosted_url: tokens.response['hosted_url']}
```
Comment (andrewsli) [inline lib/berbix.js:177]: I feel like I could/should create a class for this response instead of custom building the object here.

**Review by ChristianHansen — COMMENTED** [review]
Just slacked you another idea

File: README.md [hunk 2]
```diff
@@ -65,10 +65,14 @@ Supported options:
...
 - `customerUid` - An ID or identifier for the user in your system.
 - `templateKey` - The template key for this transaction.
-- `hostedOptions` - Optional configuration object for creating hosted transactions. The `hostedOptions` object can optionally include the following fields:
+- Deprecated: `hostedOptions` - Optional configuration object for creating hosted transactions. The `hostedOptions` object can optionally include the following fields:
   - `completionEmail` - Email address to which completion alerts will be sent for this transaction.
   - `redirectUrl` - URL to redirect the user to after they complete the transaction. If not specified, the URL specified in the Berbix dashboard will be used instead.
 
+##### `createHostedTransaction(options : object): object`
+
+Creates a hosted transaction within Berbix to initialize the client SDK. This works the same as create_transaction except that it returns an explicit `hosted_url` for hosted transactions.
```
Comment (ChristianHansen) [inline README.md:74]:
```suggestion
Creates a hosted transaction within Berbix to initialize the client SDK. This works the same as `createTransaction()` except that the object returned includes an explicit `hostedUrl` property for hosted transactions.
```

File: lib/berbix.js [hunk 3]
```diff
@@ -159,35 +171,45 @@ Client.prototype._fetchTokens = async function (path, payload) {
...
 };
 
-// createUser is deprecated, please use createTransaction
-Client.prototype.createUser = Client.prototype.createTransaction;
+Client.prototype.createHostedTransaction = function (opts) {
+  var payload = optionsToPayloadObject(opts);
+  var txnPromise = this._createTransaction("/v0/transactions", payload);
+  async function hostedResp() {
+    var txnRes = await txnPromise
+    return {tokens: txnRes.tokens, hosted_url: txnRes.hostedProperties["hosted_url"]}
```
Comment (ChristianHansen) [inline lib/berbix.js:209]:
```suggestion
    return {tokens: txnRes.tokens, hostedUrl: txnRes.hostedProperties["hosted_url"]}
```

Nit: camelCase for consistency with other props in the SDK interface

**Review by ChristianHansen — COMMENTED** [review]
two nits, otherwise looks good

**Review by jnnnthnn — APPROVED** [review]

**Pull request merged**